c/include/foxglove-c/foxglove-c.h

```diff
@@ -16,33 +16,82 @@
 #include <stdlib.h>
 
 #ifndef FOXGLOVE_NONNULL
-#if defined(__clang__)
-#define FOXGLOVE_NONNULL __attribute__((nonnull))
-#else
 #define FOXGLOVE_NONNULL
 #endif
-#endif
 
 
 typedef struct foxglove_channel foxglove_channel;
 
 
+/**
+ * Allow clients to advertise channels to send data messages to the server.
+ */
+#define FOXGLOVE_SERVER_CAPABILITY_CLIENT_PUBLISH (1 << 0)
+
+/**
+ * Allow clients to subscribe and make connection graph updates
+ */
+#define FOXGLOVE_SERVER_CAPABILITY_CONNECTION_GRAPH (1 << 1)
+
+/**
+ * Allow clients to get & set parameters.
+ */
+#define FOXGLOVE_SERVER_CAPABILITY_PARAMETERS (1 << 2)
+
+/**
+ * Inform clients about the latest server time.
+ *
+ * This allows accelerated, slowed, or stepped control over the progress of time. If the
+ * server publishes time data, then timestamps of published messages must originate from the
+ * same time source.
+ */
+#define FOXGLOVE_SERVER_CAPABILITY_TIME (1 << 3)
+
+/**
+ * Allow clients to call services.
+ */
+#define FOXGLOVE_SERVER_CAPABILITY_SERVICES (1 << 4)
+
 typedef struct foxglove_websocket_server foxglove_websocket_server;
 
+typedef struct foxglove_client_channel {
+  uint32_t id;
+  const char *FOXGLOVE_NONNULL topic;
+  const char *FOXGLOVE_NONNULL encoding;
+  const char *FOXGLOVE_NONNULL schema_name;
+  const char *schema_encoding;
+  const void *schema;
```

Comment:
Do people use void * in C instead of byte * ?

Reply:
As far as I know C doesn't have a byte type (only C++ has std::byte) - so yes.  As one example, `memcpy` is 
```
void *memcpy(void *restrict dst, const void *restrict src, size_t n);
```